contracts/RealmDiamond/facets/AlchemicaFacet.sol

```diff
@@ -297,6 +336,28 @@ contract AlchemicaFacet is Modifiers {
     emit ChannelAlchemica(_realmId, _gotchiId, channelAmounts, rate, radius);
   }
 
+  function getDecodedGotchiUserRoleData(
+    uint256 _gotchiId,
+    address _grantee
+  ) public returns (ProfitSplit memory profitSplit, address thirdParty) {
+    address _grantor = AavegotchiDiamond(s.aavegotchiDiamond).ownerOf(_gotchiId);
+
+    RoleData memory _roleData = IERC7432(s.rolesRegistry).roleData(LibGotchiRoles.GOTCHIVERSE_PLAYER, s.aavegotchiDiamond, _gotchiId, _grantor, _grantee);
+    return abi.decode(_roleData.data, (ProfitSplit, address));
```

Comment:
It's missing the `canChannelAlchemica` input, which should also be encoded.

When `canChannelAlchemica` is `false`, the function `channelAlchemica` should revert. I believe this is already being done somewhere using the old GotchiLending
```suggestion
    return abi.decode(_roleData.data, (bool, ProfitSplit, address));
```